core/src/main/java/com/linecorp/armeria/common/ContextStorageThreadLocal.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common;
+
+import static java.util.Objects.requireNonNull;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.util.UnstableApi;
+
+import io.netty.util.concurrent.FastThreadLocal;
+import io.netty.util.internal.InternalThreadLocalMap;
+
+/**
+ * A {@link ContextStorage} that uses thread-local to store {@link RequestContext}.
+ * Override this.
```

Comment:
Instead of exposing this class let's return it from `static ContextStorage default()`. A user can hook by implementing a storage that delegates to the default.

Reply:
Ah that's a good suggestion. Thanks!